server/src/main/java/org/cloudfoundry/identity/uaa/db/beans/FlywayConfiguration.java

```diff
@@ -63,6 +63,9 @@ public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata)
         public Flyway flyway(Flyway baseFlyway) {
             baseFlyway.repair();
             baseFlyway.migrate();
+            org.apache.tomcat.jdbc.pool.DataSource ds =
```

Comment:
do you see the leakage for all UAA starts or only in case a migration takes place / is needed ?

Reply:
Only in migrations so far ; both here and in [mysql/V4_9_2__AddPrimaryKeysIfMissing.java](https://github.com/cloudfoundry/uaa/pull/3186/commits/1d58d2a50157c8675595e5b82494d419d2cf6254#diff-fec9a569831747f5830030ab9802a3ee8e65b07cdd0a8ae8a7ed5fa808d1b19c)

Reply:
@strehle So I'm not convinced it is a leak as much as "parallelism" causes too many idle connections and overwhelms a developer postgresql instance (max=100). When I'm investigating, I don't see an actual leak in this method.

the `purge` simply instructs the pool to disconnect existing connections that aren't being used, and let the DB connection pool build up again. That way the test suite stays under the limit of max connections (which were mostly idle connections)